### PR TITLE
[tools] Fix not utility to work on iOS

### PR DESCRIPTION
**Pull request by tarunprabhu**
This addresses issue #77137 (std::system is not available on iOS). The implementation uses posix_spawn on iOS but std::system elsewhere.

This has only been tested on Linux on X86 and an older AArch64. 

@jroelofs, could you try this on an iOS and let me know if it works there?

### CODE REVIEW

File: tools/not.cpp
Comment (yi-wu-arm) [inline tools/not.cpp:20]: If I remember correctly `spawn.h` and is not a standard library, and not supported on Windows? you may not have access to `wait.h` on Windows as well... I saw you used `waitpid` only on IOS ifdef, so could you move it under `ifdef __APPLE__` as well?

**Review by tblah — COMMENTED** [review]
Changes look sensible to me, assuming that posix_spawn and waitpid on iOS are standards compliant. However, I have no experience writing software for iOS. Please wait for approval by somebody who can test on iOS.

File: tools/not.cpp
Comment (yi-wu-arm) [inline tools/not.cpp:69]: could you put a NULL instead of an environ?

**Comment by tblah** [comment]
As posix_spawn is part of POSIX, it is probably reasonably well available on other platforms (although I don't know about Windows). Perhaps we could get away without the #ifdef here. I don't feel strongly about this.

**Review by yi-wu-arm — COMMENTED** [review]
Thanks for fixing this! I am not very familiar with iOS process calls.

File: tools/not.cpp
Comment (tarunprabhu) [reply tools/not.cpp]: Done

File: tools/not.cpp
Comment (tarunprabhu) [reply tools/not.cpp]: Done

File: tools/not.cpp
Comment (jroelofs) [inline tools/not.cpp:29]:
```suggestion
#include <sys/wait.h>
```

**Review by jroelofs — APPROVED** [review]
LGTM with an include path fix.

**Comment by jroelofs** [comment]
thank you!

**Pull request merged**
